torchao/_models/llama/model.py

```diff
@@ -197,15 +197,17 @@ def forward(self, idx: Tensor, input_pos: Optional[Tensor] = None) -> Tensor:
 
         if input_pos is None: 
             mask = None
+            input_pos = torch.arange(0, idx.shape[1], device=idx.device)
```

Comment:
Thank you for your reply. It makes sense to separate the logic for creating `mask`, `input_pos`, `freq_cis` like you did here. However, creating `input_pos` here (for training case) is now not needed?

Reply:
I've also tested the inference case where `input_pos` is `None`. Is `input_pos` a required argument for inference mode?

Reply:
> inference case where input_pos is None

This is what I mentioned earlier. I think right now there is no code that uses inference w/o `input_pos`, and your auto-round PR doesn't seem to need it also. I think it's fine to support this case. Perhaps others can have other comments.

Can you update the PR description to describe the problem this PR fixes clearer? i.e. when `input_pos` is None (during training), `freq_cis` is overridden by line xxx. I will approve the PR once you added tests for training mode (`model.setup_caches(training=True)`) and add a short docstring/comment in `model.forward()`.